src/main/kotlin/ai/digital/integration/server/deploy/internals/cluster/operator/AwsEksHelper.kt

```diff
@@ -1,24 +1,391 @@
+                            "ParameterKey=ClusterName,ParameterValue=${awsEksProvider.clusterName.get()} " +
+                            "ParameterKey=NodeGroupName,ParameterValue=${awsEksProvider.nodeGroupName.get()} " +
+                            "ParameterKey=KeyName,ParameterValue=${awsEksProvider.sshKeyName.get()} " +
+                            "ParameterKey=FileSystemName,ParameterValue=${awsEksProvider.fileSystemName.get()} " +
+                            "ParameterKey=NodeDesiredSize,ParameterValue=${awsEksProvider.clusterNodeCount.get()} " +
+                            "ParameterKey=KubernetesVersion,ParameterValue=${awsEksProvider.kubernetesVersion.get()} " +
+                            "--on-failure DELETE " +
+                            "--output text",
+                    logOutput = true)
+            verifyClusterStatus(1500000, 300000, stackId, "CREATE_COMPLETE")// delay time 25 mins and sleep time 5 mins
+        }
+    }
+
+    private fun existingStack(): Boolean {
+        val stackStatusCmd = "aws --region ${getProvider().region.get()} " +
+                "cloudformation  describe-stacks" +
+                " --stack-name " +
+                "${getProvider().stack.get()} " +
+                "--query 'Stacks[0].StackStatus' " +
+                "--output text"
+
+        val result = ProcessUtil.executeCommand(project,
+                "$stackStatusCmd",
+                logOutput = true,
+                throwErrorOnFailure = false)
+        if (result == "CREATE_COMPLETE") {
+            return true
+        } else if (result.contains("does not exist")) {
+            return false
+        }
+        return false
+    }
+
+
+    private fun verifyClusterStatus(totalTimeInSec: Int, sleepTime: Long, stackId: String, status: String) {
+        val awsEksProvider: AwsEksProvider = getProvider()
+        stackStatus(awsEksProvider, totalTimeInSec, sleepTime, stackId, status)
+    }
+
+    private fun stackStatus(awsEksProvider: AwsEksProvider, totalTimeInSec: Int, sleepTime: Long, stackId: String, status: String) {
+        val stackStatusCmd = "aws --region ${awsEksProvider.region.get()} " +
+                "cloudformation  list-stacks " +
+                "--query 'StackSummaries[?StackId==`$stackId`].StackStatus' " +
+                "--output text"
+        val stackStatus = wait(status,
+                stackStatusCmd,
+                "Stack",
+                totalTimeInSec,
+                sleepTime)
+        if (!stackStatus) {
+            throw RuntimeException("Resource STACK status is not $status")
+        }
+    }
+
+    private fun wait(status: String, command: String, resource: String, totalTimeInSec: Int = 3000, sleepTime: Long = 1000): Boolean {
+        val expectedEndTime = System.currentTimeMillis() + totalTimeInSec // 20 mins
+        while (expectedEndTime > System.currentTimeMillis()) {
+            val result = ProcessUtil.executeCommand(project,
```

Comment:
![Screenshot 2021-12-15 at 10 24 20](https://user-images.githubusercontent.com/2265094/146159806-c6471a00-cd99-468c-b3e3-4a17ec3a825e.png)

Please check all warnings what Intellij IDEA shows. 
There is an explanation what is wrong with yellow code.
We should aim to have no warnings.

Reply:
done